examples/3-adapter_advanced.ipynb

```diff
@@ -0,0 +1,740 @@
+    "class XGBoostTransformer(BaseTransformer):\n",
+    "    def __init__(self, xgb_params, num_boost_round):\n",
+    "        self.estimator = None\n",
+    "        self.xgb_params = xgb_params\n",
+    "        self.num_boost_round = num_boost_round\n",
+    "        \n",
+    "    def fit(self, X, y):\n",
+    "        tr_mat = xgboost.DMatrix(X, label=y)\n",
+    "        evals = [(tr_mat, 'train')]\n",
+    "        self.estimator = xgboost.train(self.xgb_params,\n",
+    "                                       tr_mat,\n",
+    "                                       num_boost_round=self.num_boost_round,\n",
+    "                                       verbose_eval=False,\n",
+    "                                       evals=evals)\n",
+    "        return self\n",
+    "\n",
+    "    def transform(self, X, **kwargs):\n",
+    "        test_mat = xgboost.DMatrix(X)\n",
+    "        y_proba  = self.estimator.predict(test_mat)\n",
+    "        return {'y_proba': y_proba}\n",
+    "    \n",
+    "    def save(self, filepath):\n",
+    "        joblib.dump({'estimator': self.estimator,\n",
+    "                     'xgb_params': self.xgb_params,\n",
+    "                     'num_boost_round': self.num_boost_round},\n",
+    "                    filepath)\n",
+    "        \n",
+    "    def load(self, filepath):\n",
+    "        d = joblib.load(filepath)\n",
+    "        self.estimator = d['estimator']\n",
+    "        self.xgb_params = d['xgb_params']\n",
+    "        self.num_boost_round = d['num_boost_round']\n",
+    "        return self\n",
+    "    \n",
+    "def get_xgb_params():\n",
+    "    return {\n",
+    "        'objective': 'multi:softprob',\n",
+    "        \"num_class\": 10,\n",
+    "        'eta': 0.5,\n",
+    "        'max_depth': 4,\n",
+    "        'silent': True,\n",
+    "        'nthread': -1,\n",
+    "        'lambda': 1.0,\n",
+    "        'eval_metric': [\"mlogloss\", \"merror\"]\n",
+    "    }\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step = Step(name='xgboost',\n",
+    "               transformer=XGBoostTransformer(xgb_params=get_xgb_params(), num_boost_round=5),\n",
+    "               input_steps=[rename_step],\n",
+    "               cache_dirpath=CACHE_DIR,\n",
+    "               force_fitting=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step.fit_transform(data_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All right, so now we have two models trained, but we haven't checked so far how do they perform. Let's do it now. We will use one of data scientists' favourite measures: log-loss."
```

Comment:
Do we really have to use log-loss? It seems to complicate things and distract from the main point.